server/src/thirdparty/slackAPI/service.js

```diff
@@ -0,0 +1,162 @@
+import axios from 'axios';
+import env from '../env';
+import config from './config';
+import ThirdPartyApp from '../models/ThirdPartyApp';
+import tpEventEmitter, { generateActionUpdateEvent, generateAppAddedEvent, generateAppUpdatedEvent } from '../event';
+
+const getAuthUrl = (code) => `${config.api}/api/oauth.access?client_id=${env.slackClientId}&client_secret=${env.slackClientSecret}&code=${code}&redirect_uri=${encodeURIComponent(`${env.baseUri}/tp/slack/auth/redirect`)}`;
+
+const getIntegUrl = (state) => `${config.api}/oauth/authorize?scope=bot,channels:history,channels:read,chat:write:bot&client_id=${env.slackClientId}&redirect_uri=${encodeURIComponent(`${env.baseUri}/tp/slack/auth/redirect`)}&state=${state}`;
+
+const generateAuthToken = (code) => {
+  const authPromise = new Promise((resolve, reject) => {
+    axios.get(getAuthUrl(code)).then((response) => {
+      resolve(response.data);
+    }).catch((error) => {
+      reject(error);
+    });
+  });
+  return authPromise;
+};
+
+const getIntegrationId = async (userId, teamId) => {
+  const result = await ThirdPartyApp
+    .findOne({
+      'userDetails.team_id': teamId,
+      'userDetails.user_id': userId,
+    })
+    .select({ _id: 1 }).exec();
+
+  return result === null ? null : result._id;
```

Comment:
what is result is an empty object? 
It won't be null, so it will try to return id and you'll run into error. A better condition can be
`result === null || !result.id`

Reply:
Done